### PR TITLE
Add demangling and directive hiding for asm

### PR DESCRIPTION
**Pull request by chandlore**
Implements symbol demangling and the option to hide assembler directives (e.g. ".loc 2 338 0 prologue_end") from the generated assembly.  Both options default to on and are editable using the Config dialog.

**New - Demangled with Directives Removed**:
```asm
playground::main:
	sub	rsp, 56
	lea	rax, [rip + ref.2]
```

**Existing - Mangled with Directives**:
```asm
	.text
	.intel_syntax noprefix
	.file	"playground.cgu-0.rs"
	.section	.text._ZN10playground4main17hf4c1d0e5656172f7E,"ax",@progbits
	.p2align	4, 0x90
	.type	_ZN10playground4main17hf4c1d0e5656172f7E,@function
_ZN10playground4main17hf4c1d0e5656172f7E:
	.cfi_startproc
	sub	rsp, 56
.Lcfi0:
	.cfi_def_cfa_offset 64
	lea	rax, [rip + ref.2]
```

### CODE REVIEW

File: ui/frontend/index.scss [hunk 1]
```diff
@@ -427,7 +427,7 @@ $PHI: 2.61803398875;
         margin-top: 10em;
 
         width: 600px;
-        height: 600px / $PHI;
+        height: 950px / $PHI;
```
Comment (shepmaster) [inline ui/frontend/index.scss]: Why is this needed?

File: ui/src/sandbox.rs [hunk 2]
```diff
@@ -542,6 +567,7 @@ mod test {
     fn basic_functionality() {
         let req = ExecuteRequest {
             channel: Channel::Stable,
+            crate_type: CrateType::Binary,
```
Comment (shepmaster) [inline ui/src/sandbox.rs]: Oops, I guess I don't run these tests in CI at all! I should fix that...

File: ui/frontend/index.scss (hunk 1 above)
Comment (chandlore) [reply ui/frontend/index.scss]: I found that the background to the config popup would not fit the new rows, causing the "Done" button to be overlaid on top of the drop-downs as in issue #222.  I should probably have removed the divide by $PHI as the golden ratio is pretty well shot by this change.

Comment (chandlore) [reply ui/frontend/index.scss]: No issues with Intel vs AT&T, directives have the same format in both styles.  The [remove_assembler_directives](https://github.com/chandlore/rust-playground/blob/ec3518ca5887b75fc5ba2cb8d6982ac028eb3a20/ui/src/asm_cleanup.rs#L5-L19) function just searches for any lines that start with a period and excludes them from the updated string.

File: ui/src/sandbox.rs (hunk 2 above)
Comment (chandlore) [reply ui/src/sandbox.rs]: On my local test sandbox::test::memory_usage_is_limited is failing with:

```rust
---- sandbox::test::memory_usage_is_limited stdout ----
thread 'sandbox::test::memory_usage_is_limited' panicked at 'assertion failed: resp.stderr.contains("Killed")', sandbox.rs:848:8
```
Not sure if that's a real issue, or just differences between the containers on my machine and production.

**Comment by shepmaster** [comment]
Wonderful, thank you very much!
